city/routers.py

```diff
@@ -0,0 +1,48 @@
+from fastapi import APIRouter, Depends, Response, status
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from settings import settings
+from city import schemas, crud
+from dependencies import get_db
+
+
+prefix = settings.API_PREFIX
+router = APIRouter()
+
+
+@router.get(f"{prefix}/cities/", response_model=list[schemas.City])
+async def read_cities(db: AsyncSession = Depends(get_db)):
+    return await crud.get_all_cities(db=db)
+
+
+@router.post(
+    f"{prefix}/cities/",
+    response_model=schemas.City,
+    status_code=status.HTTP_201_CREATED
+)
```

Comment:
The use of 'int | None' for the 'city_id' parameter requires Python 3.10 or later. Ensure that your environment supports this syntax or use 'Optional[int]' from the 'typing' module for compatibility with earlier versions.